src/push_service/keys.rs

```diff
@@ -60,11 +61,14 @@ impl PushService {
 
     pub async fn get_pre_key(
         &mut self,
-        destination: &ServiceAddress,
+        destination: &ServiceId,
         device_id: u32,
     ) -> Result<PreKeyBundle, ServiceError> {
-        let path =
-            format!("/v2/keys/{}/{}?pq=true", destination.uuid, device_id);
+        let path = format!(
+            "/v2/keys/{}/{}?pq=true",
+            destination.raw_uuid(),
+            device_id
+        );
```

Comment:
IMO it would make sense if this had the domain separation; should we double check with Android? Matti may just know by heart.

Reply:
By domain separation, you mean for ACI and PNI?

Reply:
According to my [investigations](https://github.com/signalapp/libsignal/blob/826ee07b74666336f9eca74f2b270201ac157b37/rust/core/src/address.rs#L204), only ACI should have no prefix, and others (PNI) should have the corresponding (`pni:`) prefix.

Reply:
Yes, but does that apply to calls to these APIs endpoints?

Reply:
```java
      String path = String.format(PREKEY_DEVICE_PATH, destination.getIdentifier(), deviceSpecifier);
```
I'd say so.  I also don't find `pq=true` in the source code anymore.